File: dawaswiftdriver/src/main/java/com/dev/dawaswiftdriver/DriverService.java
package com.dev.dawaswiftdriver;

import android.annotation.SuppressLint;
import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.*;
import android.util.Log;
import androidx.annotation.Nullable;
import androidx.core.app.NotificationCompat;
import com.google.android.gms.location.*;
import com.google.gson.Gson;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.*;
import java.net.HttpURLConnection;
import java.net.Socket;
import java.net.URL;
import java.net.URLEncoder;
import java.util.HashMap;
import java.util.Map;


public class DriverService extends Service {
    LocationCallback locationCallback;
    private FusedLocationProviderClient mFusedLocationClient;
    Socket socket;
    Vibrator vibe;

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        vibe = (Vibrator) this.getSystemService(Context.VIBRATOR_SERVICE);
        return Service.START_STICKY;
    }

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        stopTracking(null);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        //return new LocationServiceBinder();
        return null;
    }

    private void initializeLocationManager() {
        if (mFusedLocationClient == null) {
            mFusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
        }
    }


    @SuppressLint("MissingPermission")
    public void startTracking(StartTrackingRequestEvent event) {
        initializeLocationManager();
        locationCallback = new LocationCallback() {
            @Override
            public void onLocationResult(LocationResult locationResult) {
                super.onLocationResult(locationResult);
            }
        };
        mFusedLocationClient.requestLocationUpdates(LocationRequest.create(), locationCallback, null);
    }

    public void stopTracking(StopTrackingRequestEvent event) {
        if (mFusedLocationClient != null) {
            try {
                mFusedLocationClient.removeLocationUpdates(locationCallback);
            } catch (Exception ignored) {

            }
        }
    }

    private Notification getNotification() {

        if(Build.VERSION.SDK_INT > 26) {
            NotificationChannel channel = new NotificationChannel("channel_01", "My Channel", NotificationManager.IMPORTANCE_DEFAULT);
            NotificationManager notificationManager = getSystemService(NotificationManager.class);
            notificationManager.createNotificationChannel(channel);
        }
        NotificationCompat.Builder builder = new NotificationCompat.Builder(getApplicationContext(), "channel_01").setAutoCancel(true);
        return builder.build();
    }


    public class LocationServiceBinder extends Binder {
        public DriverService getService() {
            return DriverService.this;
        }
    }



}

File: dawaswiftdriver/src/main/java/com/dev/dawaswiftdriver/services/DriverService.kt
package com.dev.dawaswiftdriver.services

import android.annotation.SuppressLint
import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.Service
import android.content.Context
import android.content.Intent
import android.os.Binder
import android.os.Build
import android.os.IBinder
import android.os.Vibrator
import androidx.core.app.NotificationCompat
import com.dev.dawaswiftdriver.StartTrackingRequestEvent
import com.dev.dawaswiftdriver.StopTrackingRequestEvent
import com.google.android.gms.location.FusedLocationProviderClient
import com.google.android.gms.location.LocationCallback
import com.google.android.gms.location.LocationRequest
import com.google.android.gms.location.LocationServices
import java.net.Socket


class DriverService : Service() {
    internal lateinit var locationCallback: LocationCallback
    private var mFusedLocationClient: FusedLocationProviderClient? = null
    internal var socket: Socket? = null
    internal lateinit var vibe: Vibrator

    private val notification: Notification
        get() {

            if (Build.VERSION.SDK_INT > 26) {
                val channel = NotificationChannel("channel_01", "My Channel", NotificationManager.IMPORTANCE_DEFAULT)
                val notificationManager = getSystemService(NotificationManager::class.java)
                notificationManager.createNotificationChannel(channel)
            }
            val builder = NotificationCompat.Builder(applicationContext, "channel_01").setAutoCancel(true)
            return builder.build()
        }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        vibe = this.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
        return Service.START_STICKY
    }

    override fun onDestroy() {
        super.onDestroy()
        stopTracking(null)
    }

    override fun onBind(intent: Intent): IBinder? {
        //return new LocationServiceBinder();
        return null
    }

    private fun initializeLocationManager() {
        if (mFusedLocationClient == null) {
            mFusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
        }
    }


    @SuppressLint("MissingPermission")
    fun startTracking(event: StartTrackingRequestEvent) {
        initializeLocationManager()
        locationCallback = object : LocationCallback() {
        }
        mFusedLocationClient!!.requestLocationUpdates(LocationRequest.create(), locationCallback, null)
    }

    fun stopTracking(event: StopTrackingRequestEvent?) {
        if (mFusedLocationClient != null) {
            try {
                mFusedLocationClient!!.removeLocationUpdates(locationCallback)
            } catch (ignored: Exception) {

            }

        }
    }


    inner class LocationServiceBinder : Binder() {
        val service: DriverService
            get() = this@DriverService
    }


}